src/modules/rabbitmq/state-changes/state.changes.consumer.ts

```diff
@@ -0,0 +1,380 @@
+                {
+                    context: StateChangesConsumer.name,
+                },
+            );
+
+            for (const change of dataTrieChanges) {
+                if (change.version === 0) {
+                    this.logger.warn(`Unsupported dataTrieChanges version 0`, {
+                        context: StateChangesConsumer.name,
+                    });
+                    continue;
+                }
+
+                const keyHex = Buffer.from(change.key, 'base64').toString(
+                    'hex',
+                );
+
+                if (!Object.keys(storageToFieldMap).includes(keyHex)) {
+                    this.logger.info(
+                        `Skipping key ${keyHex} ${Buffer.from(
+                            change.key,
+                            'base64',
+                        ).toString()}`,
+                        {
+                            context: StateChangesConsumer.name,
+                        },
+                    );
+                    continue;
+                }
+
+                pairUpdates[storageToFieldMap[keyHex].outputField] =
+                    storageToFieldMap[keyHex].decode(
+                        Buffer.from(change.val, 'base64'),
+                    );
+            }
+        }
+
+        return pairUpdates;
+    }
+
+    async updatePairsAndTokens(): Promise<void> {
+        this.filterAddresses = [];
+        this.pairs = new Map();
+        this.tokens = new Map();
+
+        const [pairs, tokens] = await Promise.all([
+            this.pairPersistence.getPairs(
+                {},
+                {
+                    address: 1,
+                    firstTokenId: 1,
+                    firstTokenPrice: 1,
+                    firstTokenPriceUSD: 1,
+                    firstTokenLockedValueUSD: 1,
+                    secondTokenId: 1,
+                    secondTokenPrice: 1,
+                    secondTokenPriceUSD: 1,
+                    secondTokenLockedValueUSD: 1,
+                    liquidityPoolTokenId: 1,
+                    liquidityPoolTokenPriceUSD: 1,
+                    lockedValueUSD: 1,
+                    info: 1,
+                    state: 1,
+                    totalFeePercent: 1,
+                    specialFeePercent: 1,
+                },
+            ),
+            this.tokenPersistence.getTokens(
+                {},
+                {
+                    identifier: 1,
+                    decimals: 1,
+                    price: 1,
+                    derivedEGLD: 1,
+                    liquidityUSD: 1,
+                    type: 1,
+                    pairAddress: 1,
+                },
+            ),
+        ]);
+
+        pairs.forEach((pair) => {
+            this.filterAddresses.push(pair.address);
+            this.pairs.set(pair.address, pair);
+        });
+
+        tokens.forEach((token) => this.tokens.set(token.identifier, token));
+    }
+
+    private getPairDecoders(
```

Comment:
You can refactor from this method with all the other from below and move them into a different file.